### ISSUE TITLE
[Native] Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.

### ISSUE THREAD

**Issue by dbertella** [styled-components/styled-components#766]
## Version
styled-components: 1.4.6
react-native: 0.44.0
## Reproduction
This issue was reported already in #674 but it was considered fix there.  
I used this repo to have a simple example:  
https://github.com/dbertella/RNstyledPlayground

## Steps to reproduce
Declaring a functional component ex:
```
// Container.js
export default (props) => <View {...props} />
```
Compose a styled-components with it ex
```
//index.ios.js
import Container from './Container';
...
const ContainerWithBorder = styled(Container)`
  borderBottomWidth: 1
`;

...
  render() {
    return (
      <ContainerWithBorder>
        <Text>This is my container</Text>
      </ContainerWithBorder>
    );
  }
```

## Expected Behavior
I'm not using ref or innerRef anywhere, I'm not expecting this warning

## Actual Behavior
It seems to happen when you have the js debugger active.  
Stack trace:  
```
Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.

Check the render method of `Styled(undefined)`.
    in Unknown (created by Styled(undefined))
    in Styled(undefined) (created by RNstyledComponent)
    in RNstyledComponent
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
```
![screen shot 2017-05-04 at 11 35 09](https://cloud.githubusercontent.com/assets/2563196/25700053/c4856e66-30bd-11e7-88a6-30e18e60c71c.png)

**Comment by philpl** [comment]
This seems to only be affecting v1.x: https://github.com/styled-components/styled-components/blob/master/src/models/StyledNativeComponent.js#L117

It should be fixed in v2, but we need to make sure not to break it in #764 

I'm not sure whether it's worth it to still fix it for v1

**Comment by dbertella** [comment]
The workaround as I mention in the other issue is to compose always a class component.
It might not be a problem then, ex:
```
class C extends React.Component {
  props: Props;
  render() {
    return <Container {...props} />;
  }
}

export default C;
```

**Comment by philpl** [comment]
Ok, since we won't go back fixing warnings in v1 since v2 is imminent (as it has been for a few weeks — sorry about that), we'll leave this up to #764 for now

**Comment by dmueller39** [comment]
@philpl This needs to be reopened IMO:

https://github.com/MatthieuLemoine/styled-components/blob/master/src/models/StyledNativeComponent.js#L117

Passes a ref to the target regardless of whether the target is a class or a stateless function.

**Comment by dmueller39** [comment]
@philp Sorry I was pointing at the wrong link:

https://github.com/styled-components/styled-components/blob/master/src/models/StyledNativeComponent.js#L123

**Comment by mxstbr** [comment]
@dmueller39 this is fixed in v2, which is a drop-in upgrade, so just run `npm install styled-components@rc` and the warning will be gone :blush: that's the final RC, so you're also not relying on pre-release code anymore, this is actually what'll be released in the very near future. (just doing release-stuff now)

**Comment by dmueller39** [comment]
No dice @mxstbr unless there is something I am missing. Is there a newer RC than 2.0.0-19 that is not public?

![screen shot 2017-05-24 at 3 50 31 pm](https://cloud.githubusercontent.com/assets/9422359/26422684/cb9e6992-4098-11e7-863c-834368836b79.png)

**Comment by mxstbr** [comment]
Oh wait, then let's reopen this and fix it for the full release. I thought @philpl said it's fixed?

**Comment by philpl** [comment]
@dmueller39 hey, sorry. Must have missed this unfortunately. Let's get it fixed :)

Do you think it's sufficient to not pass a ref if the underlying target is a stateless functional component? Or should we additionally throw if `setNativeProps` is called on the styled component that wraps a functional component?

**Comment by dmueller39** [comment]
@philpl thanks for looking into this! I was going to take a crack at it, but based on your PR I'm glad you looked instead ;)

**Comment by philpl** [comment]
@dmueller39 no worries :) hope you're happy with the change
